pkg/manifest/anaconda_installer_iso_tree.go

```diff
@@ -74,69 +77,61 @@ func NewAnacondaInstallerISOTree(buildPipeline *Build, anacondaPipeline *Anacond
 	return p
 }
 
-func (p *AnacondaInstallerISOTree) getOSTreeCommitSources() []ostree.SourceSpec {
-	if p.OSTreeCommitSource == nil {
+func (p *AnacondaInstallerISOTree) getOSTreeCommits() []ostree.CommitSpec {
+	if p.commitSpec == nil {
 		return nil
 	}
+	return []ostree.CommitSpec{*p.commitSpec}
+}
 
-	return []ostree.SourceSpec{
-		*p.OSTreeCommitSource,
+func (p *AnacondaInstallerISOTree) getContainerSpecs() []container.Spec {
+	if p.containerSpec == nil {
+		return []container.Spec{}
 	}
+	return []container.Spec{*p.containerSpec}
 }
 
-func (p *AnacondaInstallerISOTree) getOSTreeCommits() []ostree.CommitSpec {
-	if p.ostreeCommitSpec == nil {
-		return nil
+func (p *AnacondaInstallerISOTree) getContainerSources() []container.SourceSpec {
+	if p.ContainerSource == nil {
+		return []container.SourceSpec{}
+	}
+	return []container.SourceSpec{
+		*p.ContainerSource,
 	}
-	return []ostree.CommitSpec{*p.ostreeCommitSpec}
 }
 
 func (p *AnacondaInstallerISOTree) getBuildPackages(_ Distro) []string {
 	packages := []string{
 		"squashfs-tools",
 	}
 
-	if p.OSTreeCommitSource != nil {
+	if p.CommitSource != nil {
 		packages = append(packages, "rpm-ostree")
 	}
 
+	if p.ContainerSource != nil {
+		packages = append(packages, "skopeo")
+	}
+
 	if p.OSPipeline != nil {
 		packages = append(packages, "tar")
 	}
 
 	return packages
 }
 
-func (p *AnacondaInstallerISOTree) serializeStart(_ []rpmmd.PackageSpec, _ []container.Spec, commits []ostree.CommitSpec) {
-	if len(commits) == 0 {
-		// nothing to do
-		return
-	}
-
-	if len(commits) > 1 {
-		panic("pipeline supports at most one ostree commit")
+func (p *AnacondaInstallerISOTree) serializeStart(_ []rpmmd.PackageSpec, containers []container.Spec, commits []ostree.CommitSpec) {
+	if p.commitSpec != nil || p.containerSpec != nil {
+		panic("double call to serializeStart()")
 	}
-
-	p.ostreeCommitSpec = &commits[0]
 }
 
 func (p *AnacondaInstallerISOTree) serializeEnd() {
-	p.ostreeCommitSpec = nil
+	p.commitSpec = nil
+	p.containerSpec = nil
 }
 
 func (p *AnacondaInstallerISOTree) serialize() osbuild.Pipeline {
-	// If the anaconda pipeline is a payload then we need one of two payload types
-	if p.anacondaPipeline.Type == AnacondaInstallerTypePayload {
-		if p.ostreeCommitSpec == nil && p.OSPipeline == nil {
-			panic("missing ostree or ospipeline parameters in ISO tree pipeline")
-		}
-
-		// But not both payloads
-		if p.ostreeCommitSpec != nil && p.OSPipeline != nil {
-			panic("got both ostree and ospipeline parameters in ISO tree pipeline")
-		}
-	}
```

Comment:
Don't we still need these checks but with the container payload included?

Reply:
Check is back, it seems I eradicated all checks at some point and it stayed in :(